### ISSUE TITLE
Remove API calls to set application command permissions

### ISSUE THREAD

**Issue by snazzyfox** [Pycord-Development/pycord#1302]
### Summary

Setting application command permissions is no longer possible via bots

### Reproduction Steps

Start any bot that has an application command. 

### Minimal Reproducible Code

```python
import discord
bot = discord.Bot()

@bot.slash_command(guild_ids=[GUILD_ID])
async def ping(ctx):
    await ctx.respond("pong")

bot.run("TOKEN")
```

### Expected Results

The bot should start up with no errors.

### Actual Results

The bot throws the following errors during startup: `discord.errors.HTTPException: 405 Method Not Allowed (error code: 0): 405: Method Not Allowed`

### Intents

none; irrelevant

### System Information

- Python v3.10.1-final
- py-cord v2.0.0-beta
    - py-cord pkg_resources: v2.0.0b7
- aiohttp v3.8.1
- system info: Windows 10 10.0.22000

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

Earlier today, [Discord completely replaced the existing application command permission system, and deleted API endpoints to interact with them](https://discord.com/developers/docs/change-log#updated-command-permissions)

Command permissions can now only be set via Discord server UI, or via an authenticated *user*, not by the bot itself. The "permissions" and "default_permissions" fields are now obsolete, so features around managing command permissions need to be removed from this library.

**Comment by plun1331** [comment]
This error is caught and ignored in the latest commit to master.

Permissions are not obsolete, the API still allows us to set default permissions for commands, and these will be implemented in #1183.

**Comment by snazzyfox** [comment]
Sorry, I don't mean that permissions in its entirety is obsoltete. Rather, the permission model has now changed so that they are no longer part of the bot, but rather part of each server, so it's no longer possible to register permissions as part of the bot itself.
